lib/view/drawing_canvas/widgets/canvas_side_bar.dart

```diff
@@ -237,6 +242,23 @@ class CanvasSideBar extends HookWidget {
                   child: const Text('Clear'),
                   onPressed: () => undoRedoStack.value.clear(),
                 ),
+                ValueListenableBuilder<Image?>(
```

Comment:
the `ValueListenableBuilder` is not necessary since `backgroundImage` is a `useState` hook

the code below should work, maybe we should use that instead
```
TextButton(
                  onPressed: () async {
                    if (backgroundImage.value != null) {
                      backgroundImage.value = null;
                    } else {
                      backgroundImage.value = await _getImage;
                    }
                  },
                  child: Text(
                    backgroundImage.value == null ? 'Add Background' : 'Remove Background',
                  ),
                )
```

Reply:
I updated this in the recent push